### PR TITLE
[feat]add cluster usage

### PR DESCRIPTION
**Pull request by Cyber-SiKu**
image

### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

File: curvefs/src/mds/metric/metric.h [hunk 1]
```diff
@@ -59,15 +59,26 @@ class FsMountMetric {
...
     Mutex mtx_;
 
     MountPointMetric mps_;
 };
 
+struct FsUsageMetric {
+    static const std::string prefix;
+
+    std::string fsName;
+    bvar::Status<uint64_t> used;
```
Comment (h0hmj) [inline curvefs/src/mds/metric/metric.h:74]: used -> usedBytes

File: curvefs/src/mds/metric/fs_metric.cpp [hunk 2]
```diff
@@ -26,27 +26,52 @@ namespace curvefs {
...
+
+void FsMetric::DeleteFsUsage(const std::string& fsname) {
+    std::lock_guard<Mutex> lock(usageMtx_);
+
+    auto iter = usageMetrics_.find(fsname);
+    if (iter == usageMetrics_.end()) {
+        return;
+    }
+
+    usageMetrics_.erase(iter);
```
Comment (h0hmj) [inline curvefs/src/mds/metric/fs_metric.cpp:73]: just use erase? not find plus erase iter
```suggestion
    usageMetrics_.erase(fsname);
```

File: curvefs/src/mds/fs_storage.cpp [hunk 3]
```diff
@@ -163,6 +164,7 @@ FSStatusCode MemoryFsStorage::SetFsUsage(
     const std::string& fsName, const FsUsage& fsUsage) {
     WriteLockGuard writeLockGuard(fsUsedUsageLock_);
     fsUsageMap_[fsName] = fsUsage;
+    FsMetric::GetInstance().UpdateFsUsage(fsName, fsUsage);
```
Comment (h0hmj) [inline curvefs/src/mds/fs_storage.cpp:167]: SetFsUsage may be more meaningful than UpdateFsUsage? depends on you

File: curvefs/src/mds/metric/metric.h (hunk 1 above)
Comment (Cyber-SiKu) [reply curvefs/src/mds/metric/metric.h:74]: > used -> usedBytes

fix

File: curvefs/src/mds/metric/fs_metric.cpp (hunk 2 above)
Comment (Cyber-SiKu) [reply curvefs/src/mds/metric/fs_metric.cpp:73]: > just use erase? not find plus erase iter

fix

File: curvefs/src/mds/fs_storage.cpp (hunk 3 above)
Comment (Cyber-SiKu) [reply curvefs/src/mds/fs_storage.cpp:167]: > SetFsUsage may be more meaningful than UpdateFsUsage? depends on you

fix

**Review by h0hmj — APPROVED** [review]

**Comment by Cyber-SiKu** [comment]
cicheck

File: curvefs/src/mds/metric/metric.h [hunk 4]
```diff
@@ -59,15 +59,26 @@ class FsMountMetric {
...
-                           std::unique_ptr<bvar::Status<std::string>>>;
+    using MountPointMetric = std::unordered_map<std::string,
+        std::unique_ptr<bvar::Status<std::string>>>;
     // protect mps_
     Mutex mtx_;
 
     MountPointMetric mps_;
 };
 
+struct FsUsageMetric {
```
Comment (SeanHai) [inline curvefs/src/mds/metric/metric.h:70]: maybe fs capacity is also need like fs usage

Comment (Cyber-SiKu) [reply curvefs/src/mds/metric/metric.h:70]: > maybe fs capacity is also need like fs usage

fix

**Comment by Cyber-SiKu** [comment]
cicheck

File: curvefs/src/mds/fs_storage.cpp [hunk 5]
```diff
@@ -409,6 +410,7 @@ bool PersisKVStorage::LoadAllFs() {
         {
             WriteLockGuard lock(fsUsageCacheMutex_);
             fsUsageCache_[fsInfo.fsname()] = fsUsage;
+            FsMetric::GetInstance().SetFsUsage(fsInfo.fsname(), fsUsage);
```
Comment (SeanHai) [inline curvefs/src/mds/fs_storage.cpp:413]: miss to load  fs capacity

File: curvefs/src/mds/metric/metric.h [hunk 6]
```diff
@@ -59,15 +59,26 @@ class FsMountMetric {
...
+    static const std::string prefix;
+
+    std::string fsname_;
+    bvar::Status<uint64_t> usedBytes_;
+
+    explicit FsUsageMetric(const std::string& fsname)
+        : fsname_(fsname), usedBytes_(prefix + "_fs_" + fsname + "_used", 0) {}
+
+    void SetUsage(const FsUsage& usage);
```
Comment (SeanHai) [inline curvefs/src/mds/metric/metric.h:79]: miss SetCapacity?

Comment (Cyber-SiKu) [reply curvefs/src/mds/metric/metric.h:80]: > miss SetCapacity?

fix

File: curvefs/src/mds/fs_storage.cpp (hunk 5 above)
Comment (Cyber-SiKu) [reply curvefs/src/mds/fs_storage.cpp:413]: > miss to load fs capacity

fix

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Review by SeanHai — APPROVED** [review]

**Pull request merged**